crates/bevy_ecs/src/schedule/mod.rs

```diff
@@ -708,4 +708,260 @@ mod tests {
             assert!(matches!(result, Err(ScheduleBuildError::Ambiguity)));
         }
     }
+
+    mod system_stepping {
+        use super::*;
+        use ScheduleEvent::*;
+
+        // We need a ScheduleLabel to put in `ScheduleEvent` within our tests.
+        // The actual value of the label does not matter to
+        // `Schedule::handle_event()`, so we're creating one here for testing.
+        // I also don't want to have these tests dependent on bevy_app.
+        #[derive(ScheduleLabel, Clone, Debug, PartialEq, Eq, Hash)]
+        enum TestSchedule {
+            X,
+        }
+
+        fn first_system(mut order: ResMut<SystemOrder>) {
+            order.0.push(1);
+        }
+
+        fn second_system(mut order: ResMut<SystemOrder>) {
+            order.0.push(2);
+        }
```

Comment:
Can probably drop `SystemOrder`, and do something less complex for the `SystemExecutor` tests below.